src/clients/cli.rs

```diff
@@ -400,6 +402,22 @@ impl Docker for Cli {
                 WaitFor::Duration { length } => {
                     std::thread::sleep(length);
                 }
+                WaitFor::Healthcheck { delay } => loop {
+                    let container = self.inspect(id);
+                    if let Some(BollardContainerState {
+                        health: Some(health),
+                        ..
+                    }) = container.state
+                    {
+                        if health.status == Some(HealthStatusEnum::HEALTHY) {
+                            break;
+                        }
+
+                        sleep(delay);
+                    } else {
+                        panic!("Healthcheck for the container is not configured");
+                    }
```

Comment:
You should be able to write this as a match:

```suggestion
                    match container.state {
                    	Some(BollardContainerState {
                            health: Some(HealthStatusEnum::HEALTHY),
                            ..
                        }) => break,
                    	Some(BollardContainerState {
                            health: Some(HealthStatusEnum::UNHEALTHY), // TODO: CHeck if correct enum variant
                            ..
                        }) => sleep(delay),
                        Some(BollardContainerState {
                            health: None,
                            ..
                        }) => panic!("Healthcheck for the container is not configured"),
                        _ => // handle other cases
                    }
```

Reply:
Well, while I was rewriting it as a match, I realised two things:
1. I missed another case of how docker can report that the health check is not configured — it can be reported either by `ContainerState.state == None` or by `ContainerState.state.health.status == HealthStatusEnum::NONE`.
2. Match can be ugly! :) But I'm not sure how to avoid a tiny bit of repetition without it.

Which one of the following would you prefer? I like the second option, but at the same time, having to write the same `panic!` twice disappoints me.

<img width="615" alt="image" src="https://user-images.githubusercontent.com/629281/149907414-034ebb31-aeb9-4d6e-a03f-9eea97fc0b70.png">
<img width="610" alt="image" src="https://user-images.githubusercontent.com/629281/149906752-5eac542b-2507-4751-96ef-cba1001bf067.png">

Reply:
Anyway, I force-pushed the changes from the second screenshot. Feel free to disagree :)